File: 卡码/代码随想录/1/05-栈与队列/Js/02-删除字符串所有相邻重复项/LeetCode_1047.js
//1047. 删除字符串中的所有相邻重复项
const removeDuplicates = function(s) {
  const stack=new Array();
  for(let ch of s){
    if(stack.length==0||stack[stack.length-1]!=ch) stack.push(ch);
    else stack.pop();
  }
  return stack.join("");
};

File: 卡码/代码随想录/1/05-栈与队列/C++/02-删除字符串所有相邻重复项/LeetCode_1047.cpp
//1047. 删除字符串中的所有相邻重复项
#include <string>
#include <algorithm>;
using namespace std;

class Solution {
public:
  string removeDuplicates(string s) {
    stack<char> st;
    for(char ch:s){
      if(st.empty()||st.top()!=ch) st.push(ch);
      else st.pop();
    }
    string res="";
    while(!st.empty()){
      res+=st.top();
      st.pop();
    }
    reverse(res.begin(),res.end());
    return res;
  }
};
